### PR TITLE
feat(languages): jsonnet

### PR DESCRIPTION
**Pull request by matoous**
Add jsonnet language and syntax highlight. The best tree-sitter grammar as of now seems to be: https://github.com/sourcegraph/tree-sitter-jsonnet (used in the PR).

### CODE REVIEW

File: runtime/queries/jsonnet/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"if" @conditional
+[
+  (local)
+  "function"
+] @keyword
+(comment) @comment
+
+(string) @string
+(number) @number
```
Comment (the-mikedavis) [inline runtime/queries/jsonnet/highlights.scm:9]:
```suggestion
(number) @constant.numeric
```

File: runtime/queries/jsonnet/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+[
+  (true)
+  (false)
+] @boolean
```
Comment (the-mikedavis) [inline runtime/queries/jsonnet/highlights.scm:13]:
```suggestion
] @constant.builtin.boolean
```

File: runtime/queries/jsonnet/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,27 @@
+"if" @conditional
```
Comment (the-mikedavis) [inline runtime/queries/jsonnet/highlights.scm:1]:
```suggestion
"if" @keyword.control.conditional
```

File: runtime/queries/jsonnet/highlights.scm (hunk 2 above)
Comment (matoous) [reply runtime/queries/jsonnet/highlights.scm:13]: Thanks for suggestions!

**Review by the-mikedavis — COMMENTED** [review]
I played around with the queries a bit and I have a few more highlights that I think look good:

```scm
"for" @keyword.control.repeat
"in" @keyword.operator
[(self) (dollar)] @variable.builtin
"assert" @keyword
(null) @constant.builtin
[
  ":"
  "::"
  ";"
  "="
] @punctuation.delimiter
```

What do you think?

File: runtime/queries/jsonnet/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,27 @@
+"if" @keyword.control.conditional
```
[the remark this answers is not in the document]
Comment (matoous) [reply runtime/queries/jsonnet/highlights.scm:1]: Nice, added, I also added `else`, does that make sense? Jsonnet uses the `if ... then ... else ...` syntax.

**Comment by matoous** [comment]
Screenshot 2022-09-07 at 09 24 46
@the-mikedavis looks better now, thanks a lot for suggestions. Some highlights still seem weird though but I can't tell if it's the `highlights.scm` file or it the grammar has some space for improvement as well 🤔

File: runtime/queries/jsonnet/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  (false)
+] @constant.builtin.boolean
+
+(binaryop) @operator
+(unaryop) @operator
+
+(id) @variable
+(param identifier: (id) @variable.parameter)
+(bind function: (id) @function)
+(fieldname) @string.special
```
[the remark this answers is not in the document]
Comment (matoous) [reply runtime/queries/jsonnet/highlights.scm]: done, thanks

**Review by the-mikedavis — COMMENTED** [review]
Oh hmm looks like that last commit rolled back some of the prior changes like `conditional` being `keyword.control.conditional` and number being `constant.numeric`. I think switching back to those scopes should improve how the highlights look.

It also looks like the parser could use some improvements too though. I don't think it's possible to robustly distinguish variables from properties like you can in JavaScript (example: `self.some_property`)

**Comment by matoous** [comment]
@the-mikedavis you are right, sorry, I somehow overwrote them when adjusting further stuff. Should be good now.

**Comment by matoous** [comment]
Screenshot 2022-09-08 at 09 21 21

Looks good now, let me know if there's anything else I should adjust 🙏

File: runtime/queries/jsonnet/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  (true)
+  (false)
+] @constant.builtin.boolean
+
+(binaryop) @operator
+(unaryop) @operator
+
+(param identifier: (id) @variable.parameter)
+(bind function: (id) @function)
+(fieldname) @string.special
```
Comment (the-mikedavis) [inline runtime/queries/jsonnet/highlights.scm:20]:
```suggestion
(fieldname (id) @variable.other.member)
```

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks @matoous!

**Pull request merged**